pkg/client/unversioned/transport.go

```diff
@@ -171,30 +171,43 @@ func NewClientCertTLSConfig(certData, keyData, caData []byte) (*tls.Config, erro
 	if err != nil {
 		return nil, err
 	}
-	certPool := x509.NewCertPool()
-	certPool.AppendCertsFromPEM(caData)
+
+	caPool := getCertPoolForCA(caData)
+
 	return &tls.Config{
 		// Change default from SSLv3 to TLSv1.0 (because of POODLE vulnerability)
 		MinVersion: tls.VersionTLS10,
 		Certificates: []tls.Certificate{
 			cert,
 		},
-		RootCAs:    certPool,
-		ClientCAs:  certPool,
+		RootCAs:    caPool,
+		ClientCAs:  caPool,
 		ClientAuth: tls.RequireAndVerifyClientCert,
 	}, nil
 }
 
 func NewTLSConfig(caData []byte) (*tls.Config, error) {
-	certPool := x509.NewCertPool()
-	certPool.AppendCertsFromPEM(caData)
 	return &tls.Config{
 		// Change default from SSLv3 to TLSv1.0 (because of POODLE vulnerability)
 		MinVersion: tls.VersionTLS10,
-		RootCAs:    certPool,
+		RootCAs:    getCertPoolForCA(caData),
 	}, nil
 }
 
+func getCertPoolForCA(caData []byte) *x509.CertPool {
```

Comment:
this should probably be something like `rootCAPool(caData []byte)` because of the special meaning of nil... definitely godoc that `nil` is returned for empty `caData`